Source: self-written
Upstream: yes
Reason: enable the use of system hunspell and quazip

# HG changeset patch
# User Jakob Nixdorf <flocke@shadowice.org>
# Date 1455727574 -3600
#      Wed Feb 17 17:46:14 2016 +0100
# Node ID 680e1d2ee22e43c0f3b66ab18fb9ebbbc85f698a
# Parent  fa2818913f4a0a9f25a6f0243e51634dae198f86
add the option to use system hunspell and quazip

diff -r fa2818913f4a -r 680e1d2ee22e spellerutility.h
--- a/spellerutility.h	Tue Feb 16 06:46:04 2016 +0100
+++ b/spellerutility.h	Wed Feb 17 17:46:14 2016 +0100
@@ -13,7 +13,12 @@
 
 #include "mostQtHeaders.h"
 
+#ifdef HUNSPELL_STATIC
 #include "hunspell/hunspell.hxx"
+#else
+#include <hunspell.hxx>
+#endif
+
 class SpellerUtility: public QObject {
 	Q_OBJECT
 
diff -r fa2818913f4a -r 680e1d2ee22e texstudio.pro
--- a/texstudio.pro	Tue Feb 16 06:46:04 2016 +0100
+++ b/texstudio.pro	Wed Feb 17 17:46:14 2016 +0100
@@ -475,13 +475,26 @@
         utilities
 }
 
-DEFINES += HUNSPELL_STATIC
-include(hunspell/hunspell.pri)
+isEmpty(USE_SYSTEM_HUNSPELL){
+  DEFINES += HUNSPELL_STATIC
+  include(hunspell/hunspell.pri)
+} else {
+  CONFIG += link_pkgconfig
+  PKGCONFIG += hunspell
+}
 
 include(qcodeedit/qcodeedit.pri)
 
-DEFINES += QUAZIP_STATIC
-include(quazip/quazip/quazip.pri)
+isEmpty(USE_SYSTEM_QUAZIP) {
+  DEFINES += QUAZIP_STATIC
+  include(quazip/quazip/quazip.pri)
+} else {
+  isEmpty(QUAZIP_LIB): QUAZIP_LIB = -lquazip
+  isEmpty(QUAZIP_INCLUDE): QUAZIP_INCLUDE = $${PREFIX}/include/quazip
+
+  INCLUDEPATH += $${QUAZIP_INCLUDE}
+  LIBS += $${QUAZIP_LIB}
+}
 
 include(pdfviewer/pdfviewer.pri)
 
